packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -101,7 +101,9 @@ const BlockComponent = forwardRef(
 			// should only consider tabbables within editable display, since it
 			// may be the wrapper itself or a side control which triggered the
 			// focus event, don't unnecessary transition to an inner tabbable.
-			if ( wrapper.current.contains( document.activeElement ) ) {
+			if (
+				isInsideRootBlock( wrapper.current, document.activeElement )
```

Comment:
For me this is a bug that is present on master too. If you want to select a block, you want to select the root block, not an inner one.